### DISCUSSION TITLE
[Q&A] 使用 coplilot-gpt4-service 时，调用插件没有后续对话？

### DISCUSSION THREAD

**Discussion by Hurray0** [lobehub/lobehub#1149] [General Question] [answered]
### 💻 Operating System

Other Linux

### 📦 Environment

Docker

### 🌐 Browser

Safari

### 🐛 Bug Description

我的环境是：
docker搭建的lobe-chat，api使用的是coplilot-gpt4-service搭建的本地代理url，使用gpt4/gpt3.5都没有什么问题。
但是当使用插件时，无论什么插件，目前都是请求了之后没有任何后续反馈了，如下图：
<img width="838" alt="image" src="https://github.com/lobehub/lobe-chat/assets/6030586/1982cb79-6a40-42b1-9ad0-5e8012ddfc70">

<img width="837" alt="image" src="https://github.com/lobehub/lobe-chat/assets/6030586/2b2aab22-ac5e-4c24-bc03-03548e09dec2">

不知道要怎么调试和解决呢？

[a comment by an automated account is not shown]

**Comment by scortLi** [comment]
@Hurray0 你选中插件了吗

[a comment by an automated account is not shown]

**Comment by Hurray0** [comment]
> @Hurray0 你选中插件了吗

是的，我每次都选中了插件。例如第一张图选的“学术”插件：
<img width="799" alt="image" src="https://github.com/lobehub/lobe-chat/assets/6030586/ed5c193a-93c3-44b5-aabb-140cd4f1e3f1">


我看这个[issue-864](https://github.com/lobehub/lobe-chat/issues/864)说是不支持第三方模型，是不是非openai的API都用不了插件的啊？

**Comment by Hurray0** [comment]
镜像和版本信息：
<img width="868" alt="image" src="https://github.com/lobehub/lobe-chat/assets/6030586/e0be0032-2e20-4fcd-856c-12a97cf81536">

**Comment by scortLi** [comment]
@Hurray0 
![image](https://github.com/lobehub/lobe-chat/assets/14088043/c9111990-9047-4233-a4b8-4016ca108cbb)
看这个是只要支持Function call是支持第三方模型的，还有看看是不是插件描述有问题

**Comment by Hurray0** [comment]
> @Hurray0 ![image](https://private-user-images.githubusercontent.com/14088043/299524608-c9111990-9047-4233-a4b8-4016ca108cbb.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDYxNTI3MzIsIm5iZiI6MTcwNjE1MjQzMiwicGF0aCI6Ii8xNDA4ODA0My8yOTk1MjQ2MDgtYzkxMTE5OTAtOTA0Ny00MjMzLWE0YjgtNDAxNmNhMTA4Y2JiLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAxMjUlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMTI1VDAzMTM1MlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTM2ZWQ4NzY4ODlmNDMxYjJhYzA4Njg0OTA5NTFhZTg2NjdhNWFhYmIzOTA0NGE0NjI2YzE2OWEzOTZkYWQwMDMmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.GoOIlnRGUvOtLN_uLuILagKwYqN4IHAOXnXJOn6nS4M) 看这个是只要支持Function call是支持第三方模型的，还有看看是不是插件描述有问题

我这边用的插件商店里的几个插件呀，而且都已经出现了“我将使用xxx工具开始查询”这样的字样？

[a comment by an automated account is not shown]

**Comment by scortLi** [comment]
@Hurray0 那应该是这个插件本身有问题的

[a comment by an automated account is not shown]

**Comment by WayneShao** [comment]
我也有一样的问题，使用的模型是 Azure 的 gpt-4-vision

![image](https://github.com/lobehub/lobe-chat/assets/13160811/518d2404-cae1-460e-9320-a443347d8158)

[a comment by an automated account is not shown]

**Comment by Hurray0** [comment]
> @Hurray0 那应该是这个插件本身有问题的

有什么推荐测试的插件和标准问题吗？我在我的环境试一下？

[a comment by an automated account is not shown]

**Comment by mushan0x0** [comment, marked as answer]
https://github.com/aaamoon/copilot-gpt4-service/issues/98#issuecomment-1883367613 coplilot-gpt4-service 不支持函数调用的

[a comment by an automated account is not shown]

**Comment by mushan0x0** [comment]
> 我也有一样的问题，使用的模型是 Azure 的 gpt-4-vision

gpt-4-vision 也不支持函数调用

[a comment by an automated account is not shown]

**Comment by Hurray0** [comment]
> [aaamoon/copilot-gpt4-service#98 (comment)](https://github.com/aaamoon/copilot-gpt4-service/issues/98#issuecomment-1883367613) coplilot-gpt4-service 不支持函数调用的

噢噢，原来是这样。感谢解答问题🙏🏻
不过能否增加一个功能给个反馈啊（比如当前API不支持function calls），不然用户调了插件没有任何反应都不知道怎么回事

[a comment by an automated account is not shown]

**Comment by mushan0x0** [comment]
> 噢噢，原来是这样。感谢解答问题🙏🏻 不过能否增加一个功能给个反馈啊（比如当前API不支持function calls），不然用户调了插件没有任何反应都不知道怎么回事

这个判断不了，只能让接口不支持的时候进行报错，但是现在接口都不会报错

[a comment by an automated account is not shown]

**Comment by WayneShao** [comment]
> > 我也有一样的问题，使用的模型是 Azure 的 gpt-4-vision
> 
> gpt-4-vision 也不支持函数调用

![image](https://github.com/lobehub/lobe-chat/assets/13160811/98a54bea-bab8-467f-8f32-12886e349a1d)

根据[官方文档](https://learn.microsoft.com/en-us/azure/ai-services/openai/how-to/function-calling) 的说明，似乎不是不支持，而是API 破坏性更新了？

**Comment by arvinxx** [comment]
@WayneShao 不是， gpt-4-vision 不支持函数调用是这个模型本身的特性…

[a comment by an automated account is not shown]
